compiler/AST/primitive.cpp

```diff
@@ -272,12 +284,21 @@ returnInfoGetMember(CallExpr* call) {
   VarSymbol* var = toVarSymbol(sym->symbol());
   if (!var)
     INT_FATAL(call, "bad member primitive");
-  if (var->immediate) {
-    const char* name = var->immediate->v_string;
-    for_fields(field, ct) {
-      if (!strcmp(field->name, name))
-        return field->qualType();
+  if (Immediate* imm = var->immediate) {
```

Comment:
Could you also remove the "if (!ct)" test above, since you are editing anyway?

Reply:
I moved it above the check for a flag on ct->symbol